backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java

```diff
@@ -0,0 +1,97 @@
+package se.chalmers.dat265.group1.api.order;
+
+import se.chalmers.dat265.group1.api.ApiController;
+import se.chalmers.dat265.group1.model.Order;
+import se.chalmers.dat265.group1.model.dbEntities.OrderData;
+import se.chalmers.dat265.group1.model.dbEntities.OrderedPart;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class OrderController extends ApiController implements OrderAPI {
+    public OrderController(boolean debug) {
+        super(debug);
+    }
+
+    @Override
+    public List<Order> getAllOrders() {
+        List<OrderData> orderDataList = orderDataRepository.getObjects();
+        return getMerged(orderDataList);
+
+    }
+
+    private List<Order> getMerged(List<OrderData> orderDataList) {
+        List<Order> orders = new LinkedList<>();
+        for (OrderData orderData : orderDataList) {
+            Order temp = new Order(orderData, orderedPartRepository.getObjects("orderID=" + orderData.getId()));
+            orders.add(temp);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order getOrder(String orderID) {
+        OrderData orderData = orderDataRepository.getObject(Integer.parseInt(orderID));
+        return new Order(orderData, orderedPartRepository.getObjects("orderID=" + orderData.getId()));
+    }
+
+    @Override
+    public Order createNewOrder(Order order) {
+        int id = orderDataRepository.postObject(new OrderData(order)).getId();
+        for (OrderedPart orderedPart : order.getOrderedParts()) {
+            orderedPart.setID(id);
+            orderedPartRepository.postObject(orderedPart);
+        }
+        return getOrder(id + "");
+    }
+
+    @Override
+    public List<OrderedPart> getOrderedParts(String orderID) {
+        return orderedPartRepository.getObjects("orderID=" + orderID);
+    }
+
+    @Override
+    public Order updateOrder(String orderID, Order order) {
+        checkIDs(orderID, order);
+
+        OrderData orderData = orderDataRepository.updateObject(new OrderData(order));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "orderData". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "orderData" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)